### PR TITLE
queryserving, observability: instrument vttablet query cache plan hits/misses

### PR DESCRIPTION
**Pull request by maxenglander**
## Description

Add stats counters:

 * `QueryCacheHits`: Query engine query cache hits
 * `QueryCacheMisses`: Query engine query cache misses

A query plan cache miss results in VTTablet parsing the query. This can add a significant amount of latency to a query execution. Having better visibility into query plan caching will help isolate or eliminate cache misses as a source of latency when analyzing query performance.
 
## Related Issue(s)

Fixes https://github.com/vitessio/vitess/issues/14946

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [ ] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added
-   [ ] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added: https://github.com/vitessio/website/pull/1663

### CODE REVIEW

**Review by harshit-gangal — APPROVED** [review]

File: changelog/19.0/19.0.0/summary.md [hunk 1]
```diff
@@ -65,6 +66,13 @@ Prior to 19.0 VTTablet reported how much time non-streaming executions spend wai
 
 The build version (e.g., `19.0.0-SNAPSHOT`) has been added to `/debug/vars`, allowing users to programmatically inspect Vitess components' build version at runtime.
 
+#### <a id="vttablet-query-cache-hits-and-misses"/>VTTablet Query Cache Hits and Misses
+
+VTTablet exposes two counter stats:
+
+ * `QueryCacheHits`: Query engine query cache hits
+ * `QueryCacheMisses`: Query engine query cache misses
+
```
Comment (frouioui) [inline changelog/19.0/19.0.0/summary.md:75]: This should go in the `changelog/20.0/20.0.0/summary.md` summary file, if it does not exist yet you can create it.

**Review by frouioui — APPROVED** [review]
LGTM but the release notes must be changed. 🥳

File: changelog/19.0/19.0.0/summary.md (hunk 1 above)
Comment (frouioui) [reply changelog/19.0/19.0.0/summary.md:75]: Also a small nit picking, reading this it seems like VTTablet is only exposing these two counter stats now, maybe add "new" or something

**Review by frouioui — APPROVED** [review]
Still LGTM

**Comment by frouioui** [comment]
Conflicts are resolved, should be merged soon!

**Pull request merged**